contracts/validators/auth/AuthV2Validator.sol

```diff
@@ -0,0 +1,246 @@
+        mapping(string => uint256) _inputNameToIndex;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("iden3.storage.AuthV2Validator")) - 1))
+    //  & ~bytes32(uint256(0xff));
+    // solhint-disable-next-line const-name-snakecase
+    bytes32 private constant AuthV2ValidatorStorageLocation =
+        0x5212d71c1540b1d75013e45246a2b44f2ee9363a102ea02fac1792932b691600;
+
+    /// @dev Get the main storage using assembly to ensure specific storage location
+    function _getAuthV2ValidatorStorage() private pure returns (AuthV2ValidatorStorage storage $) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            $.slot := AuthV2ValidatorStorageLocation
+        }
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @dev Initialize the contract
+     * @param _stateContractAddr Address of the state contract
+     * @param _verifierContractAddr Address of the verifier contract
+     * @param owner Owner of the contract
+     */
+    function initialize(
+        address _stateContractAddr,
+        address _verifierContractAddr,
+        address owner
+    ) public initializer {
+        _initDefaultStateVariables(_stateContractAddr, _verifierContractAddr, CIRCUIT_ID, owner);
+    }
+
+    /**
+     * @dev Get the version of the contract
+     * @return Version of the contract
+     */
+    function version() public pure override returns (string memory) {
+        return VERSION;
+    }
+
+    /**
+     * @dev Parse the public signals
+     * @param inputs Array of public inputs
+     * @return Parsed public signals
+     */
+    function parsePubSignals(uint256[] memory inputs) public pure returns (PubSignals memory) {
+        PubSignals memory pubSignals = PubSignals({
+            userID: inputs[0],
+            challenge: inputs[1],
+            gistRoot: inputs[2]
+        });
+
+        return pubSignals;
+    }
+
+    /**
+     * @dev Verify the groth16 proof and check the request query data
+     * @param sender Sender of the proof.
+     * @param proof Proof packed as bytes to verify.
+     * @param params Request query data of the credential to verify.
+     * @return userID user ID of public signals as result.
+     */
+    function verify(
+        // solhint-disable-next-line no-unused-vars
+        address sender,
+        bytes calldata proof,
+        // solhint-disable-next-line no-unused-vars
+        bytes calldata params
+    ) public view override returns (uint256 userID, AuthResponseField[] memory) {
+        (
+            uint256[] memory inputs,
+            uint256[2] memory a,
+            uint256[2][2] memory b,
+            uint256[2] memory c
+        ) = abi.decode(proof, (uint256[], uint256[2], uint256[2][2], uint256[2]));
+
+        PubSignals memory pubSignals = parsePubSignals(inputs);
+        _checkGistRoot(pubSignals.userID, pubSignals.gistRoot);
```

Comment:
Please check if it works cross-chain.

Reply:
Yes, it should. It is meant that `idType` + `gistRoot` will is pub to State via as an oracle data before the authV2 is called.